File: src/main/scala/io/github/jbellis/brokk/analyzer/CodeUnitType.scala
package io.github.jbellis.brokk.analyzer

import io.github.jbellis.brokk.CodeUnitType

enum CodeUnitType {
  case CLASS, FUNCTION, FIELD
}

object CodeUnitType {
  val ALL: java.util.Set[CodeUnitType] = java.util.Set.of(CLASS, FUNCTION, FIELD)
}

File: app/src/main/java/io/github/jbellis/brokk/analyzer/CodeUnitType.java
package io.github.jbellis.brokk.analyzer;

import java.util.EnumSet;
import java.util.Set;

public enum CodeUnitType {
    CLASS,
    FIELD,
    FUNCTION,
    MODULE;

    public static final Set<CodeUnitType> ALL = EnumSet.of(CLASS, FIELD, FUNCTION, MODULE);
}
